_docs/_latest/develop/dev/inventory.md

```diff
@@ -5,49 +5,170 @@ order: 102
+If the API client is not found, you will have to add it. See [cloud_sql.py](https://github.com/GoogleCloudPlatform/forseti-security/blob/stable/google/cloud/forseti/common/gcp_api/cloudsql.py)
+for a small and self-contained example.
+
+## Create iterator to retrieve the data with the API client
+
+Create iter_xyz() that will call the API client to retrieve the data.
+
+```
+google/cloud/forseti/services/inventory/base/gcp.py
+
+    @create_lazy('compute', _create_compute)
+    def iter_images(self, projectid):
+        """Image Iterator from gcp API call
+
+        Yields:
+            dict: Generator of image resources
+        """
+        for image in self.compute.get_images(projectid):
+            yield image
+```
+
+Create XYZIterator to call the iter_xyz(), and cast the result for storage
+in Inventory.
+
+```
+google/cloud/forseti/services/inventory/base/resources.py
+
+    class ImageIterator(ResourceIterator):
+        def iter(self):
+            gcp = self.client
+            if (self.resource.enumerable() and
+                   self.resource.compute_api_enabled(gcp)):
+                for data in gcp.iter_images(
+                        projectid=self.resource['projectId']):
+                    yield FACTORIES['image'].create_new(data)
+```
+
+In order to do the casting, we need to create a resource class for xyz. This
+allows us to access the `id` and `type`. If the resource has no provided `id`,
+we must synthesize our own.  See other existing `key()`, to see how to create
+a synthetic key.
+
+```
+google/cloud/forseti/services/inventory/base/resources.py
+
+    class Image(Resource):
+        def key(self):
+            return self['id']
+
+        def type(self):
+            return 'image'
+```
+
+## Add the iterator to Inventory factories to make it available for crawling
+
+Create ResourceFactory for xyz, and link the XYZIterator to the parent resource.
+
+```
+google/cloud/forseti/services/inventory/base/resources.py
+
+    FACTORIES = {
+        'project': ResourceFactory({
+            'dependsOn': ['organization', 'folder'],
+            'cls': Project,
+            'contains': [
+                ImageIterator,
+                FirewallIterator,
+                NetworkIterator,
+                SubnetworkIterator,
+            ]}),
+       'image': ResourceFactory({
+            'dependsOn': ['project'],
+            'cls': Image,
+            'contains': [
+                ]}),
+    }
+```
+
+## Add the Inventory data to data model
+
+Assumption here is that this is a simple resource, and will go into the
+resource data model table. If you need to convert the Inventory data into
+more complicated data model, please email `discuss@forsetisecurity.org` for
+further assistance.
+
+Add xyz into `gcp_type_list`, and create `_convert_xyz()` to store the inventory
```

Comment:
We've used foo and bar before, let's stick with that rather than xyz

Reply:
good idea, done